src/http11.rs

```diff
@@ -0,0 +1,311 @@
+//! Adapts the HTTP/1.1 implementation into the `HttpMessage` API.
+use std::io::{self, Write, BufWriter, Read};
+use std::net::Shutdown;
+
+use method::{Method};
+use header::{ContentLength, TransferEncoding};
+use header::Encoding::Chunked;
+
+use net::{NetworkConnector, NetworkStream, ContextVerifier};
+use http::{HttpWriter, LINE_ENDING};
+use http::HttpReader::{SizedReader, ChunkedReader, EofReader};
+use http::HttpWriter::{ChunkedWriter, SizedWriter, EmptyWriter};
+use buffer::BufReader;
+use http::{self, HttpReader};
+
+use message::{
+    Protocol,
+    HttpMessage,
+    RequestHead,
+    ResponseHead,
+};
+use header;
+use version;
+
+/// An implementation of the `HttpMessage` trait for HTTP/1.1.
+#[derive(Debug)]
+pub struct Http11Message {
+    stream: Option<Box<NetworkStream + Send>>,
```

Comment:
Instead of having 3 options (and the possible panic! states below if all are `None`), how about some sort of `MessageState` enum, with the 3 variants. I think that would be clearer that these determine the state of the message, whereas reading the 3 options at first left me confused. Additionally, we could remove the `panic!`s where all 3 options are None, since the enum will enforce for us the state.